plasmapy/diagnostics/thomson.py

```diff
@@ -542,6 +555,24 @@ def spectral_density(
     else:
         instr_func_arr = None
 
+    # Valildate notch input
+
+    if notches is not None:
+        notches_unitless = notches.to(u.m).value
+
+        if len(np.shape(notches_unitless)) == 1:
+            notches_unitless = np.array([notches_unitless])
+
+        for notch in notches_unitless:
```

Comment:
I see - you do have the exceptions + validation here. So, you can just remove them from the lite function.